src/main/java/com/gluonhq/richtext/model/PieceTable.java

```diff
@@ -588,17 +588,17 @@ protected void doRedo(PieceTable pt) {
                     if (offset > 0) {
                         additions.add(piece.pieceBefore(offset));
                     }
-                    additions.add(piece.copy(piece.start + offset, length, decoration));
+                    additions.add(piece.copy(piece.start + offset, length, decoration.normalize(piece.getDecoration())));
```

Comment:
Why the change? It doesn't make a difference if you call `normalize` before the piece copy, or if you normalize the decoration as part of the new piece call (as it was before), does it?

Reply:
I was doing some debugging and left it. This is now fixed with latest commit.